sdk-commons/src/main/java/org/openmrs/maven/plugins/utility/DistroHelper.java

```diff
@@ -299,213 +303,152 @@ public void saveDistroPropertiesTo(File destination, String distro) throws MojoE
-				} catch (IOException e) {
-					log.warn("Failed to delete temporary directory: {}", tempDirectory.getAbsolutePath(), e);
+	public void validateDistribution(DistroProperties distroProperties) throws MojoExecutionException {
+		List<MissingDependency> missingDependencies = getMissingDependencies(distroProperties);
+		if (!missingDependencies.isEmpty()) {
+			StringBuilder message = new StringBuilder();
+			for (MissingDependency d : missingDependencies) {
+				message.append("\n");
+				message.append(d.getDependentComponent()).append(" requires ").append(d.getRequiredType());
+				message.append(" ").append(d.getRequiredComponent()).append(" version ").append(d.getRequiredVersion());
+				if (StringUtils.isBlank(d.getCurrentVersion())) {
+					message.append(" but this ").append(d.getRequiredType()).append(" is not included");
+				}
+				else {
+					message.append(" but found version ").append(d.getCurrentVersion());
 				}
 			}
+			throw new MojoExecutionException(message.toString());
 		}
 	}
 
 	/**
-	 * Downloads and processes content packages specified in the given distro properties.
-	 *
-	 * <p>This method filters out properties starting with a specific prefix (defined by {@code CONTENT_PREFIX})
-	 * from the {@code distroProperties} file, identifies the corresponding versions, and downloads the
-	 * associated ZIP files from the Maven repository. It then processes each downloaded ZIP file to locate
-	 * and parse a {@code content.properties} file, ensuring that the content package is valid and meets
-	 * the expected requirements.</p>
-	 *
-	 * <p>If a {@code groupId} is overridden for a particular content package, the method uses the overridden
-	 * value when fetching the package from Maven. The ZIP files are temporarily stored and processed to extract
-	 * the {@code content.properties} file, which is then validated and compared against the dependencies specified
-	 * in the {@code distro.properties} file.</p>
-	 *
-	 * @param contentPackageZipFile The directory where content package ZIP files will be temporarily stored.
-	 * @param distroProperties The {@code DistroProperties} object containing key-value pairs that specify
-	 *                         content packages and other properties needed to build a distribution.
-	 *
-	 * @throws MojoExecutionException If there is an error during the download or processing of the content packages,
-	 *                                such as missing or invalid {@code content.properties} files, or any IO issues.
+	 * This inspects the distribution based on distro properties, and indicates any incompatibilities in the declared version
+	 * Currently, this only reviews dependencies declared within content packages, though this could be expanded to validate
+	 * modules based on config.xml, etc.
 	 */
-	public void downloadContentPackages(File contentPackageZipFile, DistroProperties distroProperties)
-			throws MojoExecutionException {
-
-		Properties contentProperties = new Properties();
-
```

Comment:
This is what actually retrieves and calculates missing dependencies.  If we wanted to add back in some interactivity with the wizard and allow the user to choose whether to upgrade their server/distribution to the necessary artifacts and versions, we would change the goal from calling the `validateDistribution` method to calling this method and then using the list of missing dependencies to drive the wizard and support the user upgrading.